src/main/java/com/blackducksoftware/integration/hub/alert/web/actions/distribution/DistributionConfigActions.java

```diff
@@ -129,6 +140,64 @@ public String validateConfig(final R restModel) throws AlertFieldException {
         return Collections.emptyList();
     }
 
+    protected void saveConfiguredProjects(final CommonDistributionConfigEntity commonEntity, final R restModel) {
+        if (Boolean.TRUE.equals(commonEntity.getFilterByProject())) {
+            final List<String> configuredProjectsFromUI = restModel.getConfiguredProjects();
+            if (configuredProjectsFromUI != null) {
+                // Remove the old DistributionProjectRelations for the distribution configuration:
+                final List<DistributionProjectRelation> distributionProjects = distributionProjectRepository.findByCommonDistributionConfigId(commonEntity.getId());
+                distributionProjectRepository.delete(distributionProjects);
+                // Add new DistributionProjectRelations (and ConfiguredProjectEntities if needed):
+                for (final String projectName : configuredProjectsFromUI) {
+                    Long projectId;
+                    final ConfiguredProjectEntity foundEntity = configuredProjectsRepository.findByProjectName(projectName);
+                    if (foundEntity != null) {
+                        projectId = foundEntity.getId();
+                    } else {
+                        final ConfiguredProjectEntity createdEntity = configuredProjectsRepository.save(new ConfiguredProjectEntity(projectName));
+                        projectId = createdEntity.getId();
+                    }
+                    distributionProjectRepository.save(new DistributionProjectRelation(commonEntity.getId(), projectId));
+                }
+                cleanUpConfiguredProjects();
+            }
+            logger.warn("{}: List of configured projects was null; configured projects will not be updated.", commonEntity.getName());
+        }
+    }
+
+    protected List<String> getConfiguredProjects(final CommonDistributionConfigEntity commonEntity) {
+        final List<DistributionProjectRelation> distributionProjects = distributionProjectRepository.findByCommonDistributionConfigId(commonEntity.getId());
+        final List<String> configuredProjects = new ArrayList<>();
+        for (final DistributionProjectRelation relation : distributionProjects) {
+            final ConfiguredProjectEntity entity = configuredProjectsRepository.findOne(relation.getProjectId());
+            configuredProjects.add(entity.getProjectName());
+        }
+        return configuredProjects;
+    }
+
+    protected void cleanUpConfiguredProjects() {
+        final List<ConfiguredProjectEntity> configuredProjects = configuredProjectsRepository.findAll();
```

Comment:
I feel there is a query we can use that will get the projects needed automatically and maybe even delete them for us, all in one go.

Reply:
https://stackoverflow.com/questions/23723025/spring-data-delete-by-is-supported